src/android/Geolocation.java

```diff
@@ -251,56 +313,158 @@ public void onLocationResultError(LocationContext locationContext, LocationError
 
         if (locationContext.getType() == LocationContext.Type.UPDATE) {
             result.setKeepCallback(true);
-        }
-        else {
+        } else {
             locationContexts.delete(locationContext.getId());
         }
 
         locationContext.getCallbackContext().sendPluginResult(result);
     }
 
-    private void requestLocationUpdatesIfSettingsSatisfied(final LocationContext locationContext, final LocationRequest request) {
+    @Override
+    public void onLocationResultSuccess(LocationContext locationContext, List<Location> locations) {
+        for (Location location : locations) {
+            onLocationResult(location, locationContext);
+        }
+    }
+
+    private void onLocationResult(Location location, LocationContext locationContext) {
+        try {
+            JSONObject locationObject = LocationUtils.locationToJSON(location);
```

Comment:
Most of the code inside the `try` statement is duplicated on the `catch`. Maybe we could clean this up?

Reply:
I got your point, really thanks for the explanation but the problem is not only inside this function, it's inside: requestLocationUpdatesIfSettingsSatisfiedGms or requestLocationUpdatesIfSettingsSatisfiedHms, because I do not have access to modify LocationRequest from Google or Huawei to implement/extend this new interface as you suggested. I totally understand your point and looks like really better but in this case, we can leave as it it.  :)